src/nat/builder/function.py

```
@@ -342,3 +345,182 @@ class FunctionImpl(LambdaFunction[input_type, streaming_output_type, single_outp
+        converters : list[Callable] | None, optional
+            The converters to use for the function.
+
+        Raises
+        ------
+        ValueError
+            When the function name is empty or blank.
+            When the function name contains invalid characters.
+            When the function already exists in the function group.
+        """
+        if not name.strip():
+            raise ValueError("Function name cannot be empty or blank")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)